src/azure-cli/azure/cli/command_modules/sql/_params.py

```diff
@@ -1476,29 +1476,133 @@ def _configure_security_policy_storage_params(arg_ctx):
+            help='If specified, updates retention days for a deleted database, instead of an existing database.'
+            'Must match the deleted time of a deleted database on the source Managed Instance.')
 
-        c.argument('retention_days',
-                   options_list=['--retention-days'],
-                   required=True,
-                   help='New backup short term retention policy in days.'
-                   'Valid policy for live database is 7-35 days, valid policy for dropped databases is 0-35 days.')
+        c.argument(
+            'retention_days',
+            options_list=['--retention-days'],
+            required=True,
+            help='New backup short term retention policy in days.'
+            'Valid policy for live database is 7-35 days, valid policy for dropped databases is 0-35 days.')
 
     with self.argument_context('sql midb short-term-retention-policy show') as c:
-        c.argument('deleted_time',
-                   options_list=['--deleted-time'],
-                   help='If specified, shows retention days for a deleted database, instead of an existing database.'
-                   'Must match the deleted time of a deleted database on the source Managed Instance.')
+        c.argument(
+            'deleted_time',
+            options_list=['--deleted-time'],
+            help='If specified, shows retention days for a deleted database, instead of an existing database.'
+            'Must match the deleted time of a deleted database on the source Managed Instance.')
+
+    with self.argument_context('sql midb long-term-retention-policy set') as c:
+        create_args_for_complex_type(
+            c, 'parameters', ManagedDatabase, [
+                'weekly_retention',
+                'monthly_retention',
+                'yearly_retention',
+                'week_of_year'
+            ])
+
+        c.argument('weekly_retention',
+                   help='The Weekly Retention.'
+                   'If just a number is passed instead of an ISO 8601 string, days will be assumed as the units.'
+                   'There is a minimum of 7 days and a maximum of 10 years.')
+
+        c.argument('monthly_retention',
+                   help='The Monthly Retention.'
+                   'If just a number is passed instead of an ISO 8601 string, days will be assumed as the units.'
+                   'There is a minimum of 7 days and a maximum of 10 years.')
+
+        c.argument('yearly_retention',
+                   help='The Yearly Retention.'
+                   'If just a number is passed instead of an ISO 8601 string, days will be assumed as the units.'
+                   'There is a minimum of 7 days and a maximum of 10 years.')
+
+        c.argument('week_of_year',
+                   help='The Week of Year, 1 to 52, to save for the Yearly Retention.')
+
+    with self.argument_context('sql midb long-term-retention-backup') as c:
+        c.argument('location_name',
```

Comment:
location is required for backup show, right? may I know why location is required but no resource group?

Reply:
No matter what, a LTR backup resource id will include a location.  

For resource group - for example, if a server is dropped, then the resources on that server are also dropped, and therefore they are all disassociated with any resource group.  However, in this case we'd still like the LTR backup to be available.

Reply:
I am fine with --location optional. Because you add this argument with `required=True`, it will be required for user. 

I am thinking if --location is not necessary to be `required`, as you can see for command `az sql ltr-backup delete` as shown here:
![image](https://user-images.githubusercontent.com/49508232/79205583-b701b200-7e70-11ea-9500-59900faf8966.png)

Reply:
added backup_id to show/delete commands (discussed on teams)